internal/stores/contracts.go

```diff
@@ -253,8 +259,22 @@ type (
 	}
 )
 
+// BeforeDelete implements a deletion hook for dbContract. This is necessary
+// because we can't delete a contract without first removing the corresponding
+// entry from the contract_sectors since that table potentially keeps a
+// reference to the contract..
+func (cs *dbContract) BeforeDelete(tx *gorm.DB) error {
```

Comment:
By using hooks we probably lose acid properties right? Are we fine with that?

Reply:
We shouldn't. The flow is
```
// begin txn
// before delete hook
// delete
// after delete hook
// commit txn
```

Reply:
cool